File: util/android/src/main/java/com/dmi/util/ResourceUtils.java
package com.dmi.util;

import android.annotation.SuppressLint;
import android.content.res.Resources;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class ResourceUtils {
    @SuppressLint("NewApi")
    public static String readTextRawResource(Resources resources,
                                             final int resourceId) {
        try (InputStream inputStream = resources.openRawResource(resourceId)) {
            try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream)) {
                try (BufferedReader bufferedReader = new BufferedReader(
                        inputStreamReader)) {
                    String nextLine;
                    final StringBuilder body = new StringBuilder();

                    while ((nextLine = bufferedReader.readLine()) != null) {
                        body.append(nextLine);
                        body.append('\n');
                    }

                    return body.toString();
                }
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: util/util-android/src/main/kotlin/com/dmi/util/ResourceUtils.kt
package com.dmi.util

import android.annotation.SuppressLint
import android.content.res.Resources
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

object ResourceUtils {
    @SuppressLint("NewApi")
    fun readTextRawResource(resources: Resources,
                            resourceId: Int): String {
        try {
            resources.openRawResource(resourceId).use { inputStream ->
                InputStreamReader(inputStream).use { inputStreamReader ->
                    BufferedReader(
                            inputStreamReader).use { bufferedReader ->
                        var nextLine: String
                        val body = StringBuilder()

                        nextLine = bufferedReader.readLine()
                        while (nextLine != null) {
                            body.append(nextLine)
                            body.append('\n')
                            nextLine = bufferedReader.readLine()
                        }

                        return body.toString()
                    }
                }
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }

    }
}
